File: Asp.NET Backend/Customer_Backend/Models/OrderDetails.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace CustomerApp.Models
{
    [Table("orderdetails")]
    public class OrderDetails
    {
        [Key]
        public int OrderDetailsId { get; set; }

        [ForeignKey("Order")]
        public int OrderId { get; set; }

        [JsonIgnore]
        public Order Order { get; set; }  // Navigation property

        [ForeignKey("ProductShopkeeper")]
        public int Spid { get; set; }

        [JsonIgnore]
        public ProductShopkeeper ProductShopkeeper { get; set; }  // Navigation property

        [Required]
        public int Qty { get; set; }

        [Required]
        public float Price { get; set; }
    }
}

File: SpringBoot Backend/ShopkeeperBackEnd/src/main/java/com/example/demo/entities/OrderDetails.java
package com.example.demo.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "orderdetails")
public class OrderDetails {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer orderdetailsid;
    
    @ManyToOne
    @JoinColumn(name = "orderid", referencedColumnName = "orderid")
    private Orders order;

    @ManyToOne
    @JoinColumn(name = "spid", referencedColumnName = "spid")
    private ProductShopkeeper productShopkeeper;

    private Integer qty;

    private Float price;

    // Getters and Setters

    public Integer getOrderdetailsid() {
        return orderdetailsid;
    }

    public void setOrderdetailsid(Integer orderdetailsid) {
        this.orderdetailsid = orderdetailsid;
    }

    public Orders getOrder() {
		return order;
	}

	public void setOrder(Orders order) {
		this.order = order;
	}

	public ProductShopkeeper getProductShopkeeper() {
        return productShopkeeper;
    }

    public void setProductShopkeeper(ProductShopkeeper productShopkeeper) {
        this.productShopkeeper = productShopkeeper;
    }

    public Integer getQty() {
        return qty;
    }

    public void setQty(Integer qty) {
        this.qty = qty;
    }

    public Float getPrice() {
        return price;
    }

    public void setPrice(Float price) {
        this.price = price;
    }
}
